### ISSUE TITLE
Route through the construction site

### ISSUE THREAD

**Issue by Ciutnik** [abrensch/brouter#753]
Hi. I don't know if this is the right place to report a routing issue, but I noticed that they may be routed along roads under construction (`highway=construction`). In this case, it's a bridge that doesn't exist yet:

https://brouter.de/brouter-web/#map=17/52.22292/21.04570/standard&lonlats=21.047578,52.223741;21.044934,52.222082

I would like to take this opportunity to thank the Authors of BRouter and wish all Users the best 🙂

**Comment by Oi-sin** [comment]
Me too I fell into this trap twice last week. The misc/profiles2/trekking.brf profile does not care about the construction=* tag and guides over bridges that are not usable. 

One of my examples: routing over the bridge over the river Werse https://brouter.de/brouter-web/#map=18/52.02399/7.68867/standard&lonlats=7.691224,51.996482;7.686052,52.039082

**Comment by afischerdev** [comment]
No reaction from the friends of the profiles.
So here are my two cents:

It's a little frustrating when you're standing in front of a bridge and can't get across. Especially when you're on foot.
If you search for the keyword construction in this repository, you'll find some discussions on this topic.

How about a possible user decision?

We offer something like this:

```
assign ignore_construction        = false  # %ignore_construction% | Activate to go over a road on construction | boolean
...
assign costfactor
...
  else if ( and highway=construction  not ignore_construction ) then 10000
```
And the user can decide whether a passage is possible or not, everybody knows the own area mostly.

**Comment by devemux86** [comment]
@afischerdev Nice idea. 👍 

To which profiles do you think we should add it?

**Comment by Oi-sin** [comment]
I don't know the background to why `construction` is ignored completely. Maybe it should be evaluated in combination with access tags. But if a road or bridge is in construction and is clearly tagged with access=no or bike=no routing over such a road almost always leads into a dead end and frustration. 
As I see it, the tagging in my example above is correct: it tells clearly, that the bridge cannot be used (https://www.openstreetmap.org/way/34184179). Being routed over such a bridge seems to be worse than not being routed there even when passing on foot or by pushing the bike might be somehow possible.

**Comment by afischerdev** [comment]
@devemux86 
In my eyes it is a problem for all routing profiles.
In the repository only these profiles handle construction

```
hiking-mountain
  else if highway=proposed|abandoned|construction then (  switch ismuddy 10  switch iswet 6 4 )

mtb
    switch    highway=construction                     10000
```
And I think we need more samples.

@Oi-sin 
Yes, you are right, but this bridge also is part of a [relation](https://www.openstreetmap.org/relation/116996#map=11/51.8904/7.9817) and the rule is `a way with a bike route relation is always passable (even with access=no)`

We have to discuss if this is still true or could be changed.

I think the biker routes are more motivated by tourism or sport than by the correctness of the road layout.

The rules are collected [here](https://github.com/abrensch/brouter/issues/62#issuecomment-895122936)

**Comment by quaelnix** [comment]
> In the repository only these profiles handle construction

[gravel.brf](https://github.com/abrensch/brouter/blob/master/misc/profiles2/gravel.brf#L275
) also assigns a `costfactor` of 10000 to `highway=construction`:

https://brouter.de/brouter-web/#map=17/52.22130/21.04689/standard&lonlats=21.047578,52.223741;21.044934,52.222082&profile=gravel

**Comment by afischerdev** [comment]
@quaelnix 
Yes, you are right. I searched for `construction` and it was not found in `gravel.brf`. 
It's not explicitly mentioned and works through the last `all other` rule. 
This means that ways with construction and sidewalks (or something like this) may also be excluded, but could be passable for pedestrians or bikers.
